### DISCUSSION TITLE
Make address labels optional

### DISCUSSION THREAD

**Discussion by rynpsc** [craftcms/cms#11410] [Ideas] [answered]
It would be nice if the Address `title` field could be made optional. Whilst a nice addition to be able to give addresses labels not all stores have the concept of address labels. Users with any saved addresses per Craft/Commerce 4 now will hit a validation error if trying to update an existing address.

**Comment by GaryReckard** [comment]
I agree. Also, I'm noticing this bit here: https://github.com/craftcms/commerce/blob/504896ab85cff3d93b0e40cfacfca6c25c4121dd/src/elements/Order.php#L2924
seems to overwrite my address titles with "Shipping Address" or "Billing Address" anyway. I've tried editing the address on an order with a custom label, and it just went back to those defaults after updating the order. 🤔

**Comment by MoritzLost** [comment]
Another +1 for this. There are a lot of situations where the label is superfluous and confusing to editors. For example, if you have an entry type that represents a location and an Addresses field that allows a single Address. The title of the entry is the name of the location, so the label isn't needed. Having to fill it out and either repeat the name of the location or enter something generic like _Address_ is awkward.

@brandonkelly Are there any technical or design constraints that prevent this? Titles are already optional for entries, so making them optional for addresses should be possible?

**Reply by MoritzLost** [reply]
@brandonkelly Any chance we can get a comment on this?

We're in the process of removing external plugins in favor of the native Addresses field. There are a couple of limitations that we can mostly live with or workaround, but this is a big one. There are only very few scenarios where the label makes sense. When a user has multiple addresses (like their home and work address) in the context of a shipping address selection, fair enough. But when you use the address field to enter a single address for a store location, event location, shop, or whatever else, the label is almost always redundant.

In those cases, we need to be able to hide the label field and either define a label format, or just allow addresses to work without a label, similar to entries used in matrix fields.

I haven't really found a way to work around this, and I'm not sure how to explain this awkwardness to clients.

**Comment by brandonkelly** [comment, marked as answer]
Just made it possible to mark the address Label field as optional from the address field layout configurator, for Craft 5.9. (b365633a914879e639c4dd53e1423d1c7e5a8c5f)

**Reply by MoritzLost** [reply]
@brandonkelly Thanks, works great! 🚀 

I noticed it's still not possible to remove the label entirely from the field layout, just to make it optional. Though I can use conditions to hide it now, and it will pass validation. Is it possible to allow the field to be removed completely, similar to the `title` field in entry types?

**Reply by brandonkelly** [reply]
Unfortunately that’s not possible right now, without having the field no longer included by default.

**Reply by MoritzLost** [reply]
@brandonkelly Ah, didn't know that was the case. Do addresses need labels to be included by default?

Feels like they're only relevant in the context of having multiple shipping addresses in a commerce system. In most of our projects, we just use the Addresses field as a structured way to store address data, and the label is almost always redundant with the owner entry …

**Reply by brandonkelly** [reply]
It’s pretty common for addresses to have labels, especially when you can manage multiple of them (e.g. within user account preferences) so I don’t see us removing them by default.

**Reply by MoritzLost** [reply]
@brandonkelly Fair enough. In that case, making it possible to have a native property be in the field layout by default, but still removable, would be on my wishlist 🙂

**Reply by brandonkelly** [reply]
Craft 5.9.0 is out with that change 🎉
